### PR TITLE
Support structured streaming writes to BigQuery

### PR DESCRIPTION
**Pull request by varundhussa**
Add a new Streaming sink
Make load requests per batch DataFrame

### CODE REVIEW

**Comment by varundhussa** [comment]
@davidrabinowitz this allows loads for each batch

File: connector/src/main/scala/com/google/cloud/spark/bigquery/RowConverter.scala [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  /**
+   * Convert InternalRow to Row
+   */
+  private val expressionEncoder: ExpressionEncoder[Row] = RowEncoder(schema).resolveAndBind()
+
+  def convertToRow(internalRow: InternalRow): Row = {
+    expressionEncoder.fromRow(internalRow)
+  }
+
+  def convert(internalRow: InternalRow): TableRow = {
```
Comment (davidrabinowitz) [inline connector/src/main/scala/com/google/cloud/spark/bigquery/RowConverter.scala:57]: This method and all those following it are not being used. Please remove them.

**Comment by davidrabinowitz** [comment]
@varundhussa Thanks for the clean up! Sorry, I forgot to ask, can you please add an integration test - as it is a new functionality I want to make sure it is properly covered.

**Comment by varundhussa** [comment]
Thanks @davidrabinowitz. I'll write a dataframe selected from BigQuery to a MemoryStream and write tests on top of it.

**Comment by varundhussa** [comment]
@davidrabinowitz I have added a append integration test. Not sure how to run it with the build process.

**Pull request merged**